### ISSUE TITLE
Bug: Sage `@asset` directive incompatible with the dist output

### ISSUE THREAD

**Issue by retlehs** [roots/bud#897]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

#### What's wrong?

Sage 10's [`@asset`](https://roots.io/docs/sage/10.x/compiling-assets/#theme-assets) directive isn't referencing assets from the correct directories

#### What have you tried?

`<img src="@asset('images/example.jpg')">`

#### What insights have you gained?

Bud's `public` directory throws the assets into an `assets/` subdirectory

#### Temporary workarounds

`<img src="@asset('assets/images/example.jpg')">`

### Steps To Reproduce

1. From a Sage 10 theme, place an image in `resources/images/`
1. Run `yarn build`
1. Try to use the `@asset` directive as documented/expected

### Expected Behavior

The asset directive should load the asset

### Actual Behavior

The asset can't be found

### Versions

v5

**Comment by kellymears** [comment]
Moving this to 5.1.0 because I think we'll end up skipping right to it since I've already merged stuff that necessitates a minor bump to the next branch.

**Comment by kellymears** [comment]
This is blocked by #891 but I've got a solve ready-to-go once it's merged.

**Comment by retlehs** [comment]
fixed in v5.1.0

ref roots/sage#2952
